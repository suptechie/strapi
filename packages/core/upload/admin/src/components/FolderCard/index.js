export { FolderCard } from './FolderCard';
export { FolderCardBody } from './FolderCardBody';
export { FolderCardCheckbox } from './FolderCardCheckbox';
export { FolderCardLink } from './FolderCardLink';
