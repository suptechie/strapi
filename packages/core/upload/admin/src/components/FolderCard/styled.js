import styled from 'styled-components';

import { BaseLink } from '@strapi/design-system/BaseLink';

export const FolderCardLink = styled(BaseLink)`
  text-decoration: none;
`;
