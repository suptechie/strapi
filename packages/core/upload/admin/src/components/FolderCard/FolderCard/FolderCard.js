import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

import { pxToRem } from '@strapi/helper-plugin';
import { Box } from '@strapi/design-system/Box';
import { CardAction } from '@strapi/design-system/Card';
import { Stack } from '@strapi/design-system/Stack';
import Folder from '@strapi/icons/Folder';

import { FolderCardContext } from '../contexts/FolderCard';
import useId from '../hooks/useId';

const FauxClickWrapper = styled.button`
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;

  &:hover,
  &:focus {
    text-decoration: none;
  }
`;

const StyledFolder = styled(Folder)`
  path {
    fill: currentColor;
  }
`;

const CardActionDisplay = styled(Box)`
  display: none;
`;

const Card = styled(Box)`
  &:hover,
  &:focus-within {
    ${CardActionDisplay} {
      display: ${({ isCardActions }) => (isCardActions ? 'block' : '')};
    }
  }
`;

export const FolderCard = ({
  children,
  id,
  startAction,
  cardActions,
  ariaLabel,
  onClick,
  to,
  ...props
}) => {
  const generatedId = useId(id);

  return (
    <FolderCardContext.Provider value={{ id: generatedId }}>
      <Card position="relative" tabIndex={0} isCardActions={!!cardActions} {...props}>
        <FauxClickWrapper
          to={to || undefined}
          as={to ? NavLink : 'button'}
          type={to ? undefined : 'button'}
          onClick={onClick}
          tabIndex={-1}
          aria-label={ariaLabel}
          aria-hidden
        />

        <Stack
          hasRadius
          background="neutral0"
          shadow="tableShadow"
          paddingBottom={3}
          paddingLeft={4}
          paddingRight={4}
          paddingTop={3}
          spacing={3}
          horizontal
          cursor="pointer"
        >
          {startAction}

          <Box
            hasRadius
            background="secondary100"
            color="secondary500"
            paddingBottom={2}
            paddingLeft={3}
            paddingRight={3}
            paddingTop={2}
          >
            <StyledFolder width={pxToRem(20)} height={pxToRem(18)} />
          </Box>

          {children}

          <CardActionDisplay>
            <CardAction right={4}>{cardActions}</CardAction>
          </CardActionDisplay>
        </Stack>
      </Card>
    </FolderCardContext.Provider>
  );
};

FolderCard.defaultProps = {
  id: undefined,
  cardActions: null,
  startAction: null,
  to: undefined,
};

FolderCard.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  id: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  startAction: PropTypes.element,
  cardActions: PropTypes.element,
  to: PropTypes.string,
};
