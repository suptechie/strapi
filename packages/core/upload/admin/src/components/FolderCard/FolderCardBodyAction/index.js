import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';

const FolderCardLink = styled(NavLink)`
  max-width: 100%;
  text-decoration: none;
`;

export const FolderCardBodyAction = ({ to, ...props }) => {
  if (to) {
    return <FolderCardLink to={to} {...props} />;
  }

  return <button type="button" {...props} />;
};

FolderCardBodyAction.defaultProps = {
  to: undefined,
};

FolderCardBodyAction.propTypes = {
  to: PropTypes.string,
};
