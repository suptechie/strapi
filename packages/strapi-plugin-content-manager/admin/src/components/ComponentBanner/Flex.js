import styled from 'styled-components';

const Flex = styled.div`
  display: flex;
  > button {
    cursor: pointer;
    padding-top: 0;
  }
  .trash-icon {
    color: #4b515a;
  }
  .button-wrapper {
    line-height: 35px;
  }
`;

export default Flex;
