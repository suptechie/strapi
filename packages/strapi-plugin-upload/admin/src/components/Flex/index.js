import styled from 'styled-components';
import PropTypes from 'prop-types';

const Flex = styled.div`
  display: flex;
  justify-content: ${({ justifyContent }) => justifyContent};
  flex-direction: ${({ flexDirection }) => flexDirection};
`;

Flex.defaultProps = {
  justifyContent: 'normal',
  flexDirection: 'row',
};

Flex.propTypes = {
  justifyContent: PropTypes.string,
  flexDirection: PropTypes.string,
};

export default Flex;
