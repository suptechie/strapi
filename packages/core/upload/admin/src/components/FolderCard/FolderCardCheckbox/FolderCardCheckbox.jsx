import React from 'react';

import { Checkbox, Box } from '@strapi/design-system';

import { useFolderCard } from '../contexts/FolderCard';

export const FolderCardCheckbox = (props) => {
  const { id } = useFolderCard();

  return (
    <Box position="relative" zIndex={2}>
      <Checkbox aria-labelledby={`${id}-title`} {...props} />
    </Box>
  );
};
