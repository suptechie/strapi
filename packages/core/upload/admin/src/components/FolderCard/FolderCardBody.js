import React from 'react';
import styled from 'styled-components';

import { Flex } from '@strapi/design-system/Flex';

import { useFolderCard } from './FolderCardContext';

const StyledBox = styled(Flex)`
  user-select: none;
`;

export const FolderCardBody = props => {
  const { id } = useFolderCard();

  return (
    <StyledBox
      {...props}
      id={`${id}-title`}
      alignItems="flex-start"
      direction="column"
      position="relative"
      zIndex={3}
    />
  );
};
