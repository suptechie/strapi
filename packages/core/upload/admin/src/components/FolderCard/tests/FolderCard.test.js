import React from 'react';
import { BaseLink } from '@strapi/design-system/BaseLink';
import { Flex } from '@strapi/design-system/Flex';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { Typography } from '@strapi/design-system/Typography';
import { render, fireEvent, act } from '@testing-library/react';

import { FolderCard } from '../FolderCard';
import { FolderCardBody } from '../FolderCardBody';
import { FolderCardCheckbox } from '../FolderCardCheckbox';

const ID_FIXTURE = 'folder';

// eslint-disable-next-line react/prop-types
const ComponentFixture = props => {
  return (
    <ThemeProvider theme={lightTheme}>
      <FolderCard
        id={ID_FIXTURE}
        ariaLabel="Folder 1"
        startAction={<></>}
        onDoubleClick={() => {}}
        {...props}
      >
        <FolderCardBody as="h2">
          <BaseLink href="https://strapi.io" textDecoration="none">
            <Flex direction="column" alignItems="flex-start">
              <Typography variant="omega" fontWeight="semiBold">
                Pictures
              </Typography>
            </Flex>
          </BaseLink>
        </FolderCardBody>
      </FolderCard>
    </ThemeProvider>
  );
};

const setup = props => render(<ComponentFixture {...props} />);

describe('FolderCard', () => {
  test('renders and matches the snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('properly calls the onDoubleClick callback', () => {
    const callback = jest.fn();
    const { container } = setup({ onDoubleClick: callback });

    act(() => {
      fireEvent(container.querySelector('button'), new MouseEvent('dblclick', { bubbles: true }));
    });

    expect(callback).toHaveBeenCalledTimes(1);
  });

  test('has all required ids set when rendering a start action', () => {
    const { container } = setup({
      startAction: <FolderCardCheckbox value={false} />,
    });

    expect(container).toMatchSnapshot();
    expect(container.querySelector(`[id="${ID_FIXTURE}-3-title"]`)).toBeInTheDocument();
    expect(
      container.querySelector(`[aria-labelledby="${ID_FIXTURE}-3-title"]`)
    ).toBeInTheDocument();
  });
});
