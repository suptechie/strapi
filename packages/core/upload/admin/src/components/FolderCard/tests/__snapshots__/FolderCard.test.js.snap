// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FolderCard has all required ids set when rendering a start action 1`] = `
.c9 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c10 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c19 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c13 {
  font-weight: 500;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c6 {
  margin: 0;
  height: 18px;
  min-width: 18px;
  border-radius: 4px;
  border: 1px solid #c0c0cf;
  -webkit-appearance: none;
  background-color: #ffffff;
  cursor: pointer;
}

.c6:checked {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c6:checked:after {
  content: '';
  display: block;
  position: relative;
  background: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iOCIgdmlld0JveD0iMCAwIDEwIDgiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPHBhdGgKICAgIGQ9Ik04LjU1MzIzIDAuMzk2OTczQzguNjMxMzUgMC4zMTYzNTUgOC43NjA1MSAwLjMxNTgxMSA4LjgzOTMxIDAuMzk1NzY4TDkuODYyNTYgMS40MzQwN0M5LjkzODkzIDEuNTExNTcgOS45MzkzNSAxLjYzNTkgOS44NjM0OSAxLjcxMzlMNC4wNjQwMSA3LjY3NzI0QzMuOTg1OSA3Ljc1NzU1IDMuODU3MDcgNy43NTgwNSAzLjc3ODM0IDcuNjc4MzRMMC4xMzg2NiAzLjk5MzMzQzAuMDYxNzc5OCAzLjkxNTQ5IDAuMDYxNzEwMiAzLjc5MDMyIDAuMTM4NTA0IDMuNzEyNEwxLjE2MjEzIDIuNjczNzJDMS4yNDAzOCAyLjU5NDMyIDEuMzY4NDMgMi41OTQyMiAxLjQ0NjggMi42NzM0OEwzLjkyMTc0IDUuMTc2NDdMOC41NTMyMyAwLjM5Njk3M1oiCiAgICBmaWxsPSJ3aGl0ZSIKICAvPgo8L3N2Zz4=) no-repeat no-repeat center center;
  width: 10px;
  height: 10px;
  left: 50%;
  top: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c6:checked:disabled:after {
  background: url(data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTAiIGhlaWdodD0iOCIgdmlld0JveD0iMCAwIDEwIDgiIGZpbGw9Im5vbmUiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CiAgPHBhdGgKICAgIGQ9Ik04LjU1MzIzIDAuMzk2OTczQzguNjMxMzUgMC4zMTYzNTUgOC43NjA1MSAwLjMxNTgxMSA4LjgzOTMxIDAuMzk1NzY4TDkuODYyNTYgMS40MzQwN0M5LjkzODkzIDEuNTExNTcgOS45MzkzNSAxLjYzNTkgOS44NjM0OSAxLjcxMzlMNC4wNjQwMSA3LjY3NzI0QzMuOTg1OSA3Ljc1NzU1IDMuODU3MDcgNy43NTgwNSAzLjc3ODM0IDcuNjc4MzRMMC4xMzg2NiAzLjk5MzMzQzAuMDYxNzc5OCAzLjkxNTQ5IDAuMDYxNzEwMiAzLjc5MDMyIDAuMTM4NTA0IDMuNzEyNEwxLjE2MjEzIDIuNjczNzJDMS4yNDAzOCAyLjU5NDMyIDEuMzY4NDMgMi41OTQyMiAxLjQ0NjggMi42NzM0OEwzLjkyMTc0IDUuMTc2NDdMOC41NTMyMyAwLjM5Njk3M1oiCiAgICBmaWxsPSIjOEU4RUE5IgogIC8+Cjwvc3ZnPg==) no-repeat no-repeat center center;
}

.c6:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c6:indeterminate {
  background-color: #4945ff;
  border: 1px solid #4945ff;
}

.c6:indeterminate:after {
  content: '';
  display: block;
  position: relative;
  color: white;
  height: 2px;
  width: 10px;
  background-color: #ffffff;
  left: 50%;
  top: 50%;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c6:indeterminate:disabled {
  background-color: #dcdce4;
  border: 1px solid #c0c0cf;
}

.c6:indeterminate:disabled:after {
  background-color: #8e8ea9;
}

.c0 {
  position: relative;
}

.c5 {
  position: relative;
  z-index: 2;
}

.c7 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
  border-radius: 4px;
}

.c12 {
  max-width: 100%;
}

.c2 {
  background: #ffffff;
  padding-top: 12px;
  padding-right: 16px;
  padding-bottom: 12px;
  padding-left: 16px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  cursor: pointer;
  cursor: pointer;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 > * {
  margin-left: 0;
  margin-right: 0;
}

.c4 > * + * {
  margin-left: 12px;
}

.c15 {
  right: 16px;
}

.c16 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c17 > * {
  margin-left: 0;
  margin-right: 0;
}

.c17 > * + * {
  margin-left: 8px;
}

.c18 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 path {
  fill: currentColor;
}

.c14 {
  display: none;
}

.c11 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      type="button"
    />
    <div
      class="c2 c3 c4"
      cursor="pointer"
      spacing="3"
    >
      <div
        class="c5"
      >
        <input
          aria-labelledby="folder-3-title"
          class="c6"
          type="checkbox"
        />
      </div>
      <div
        class="c7"
      >
        <svg
          class="c8"
          fill="none"
          height="1.125rem"
          viewBox="0 0 24 24"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.414 5H21a1 1 0 011 1v14a1 1 0 01-1 1H3a1 1 0 01-1-1V4a1 1 0 011-1h7.414l2 2z"
            fill="#212134"
          />
        </svg>
      </div>
      <h2
        class="c9 c10 c11"
        id="folder-3-title"
        overflow="hidden"
      >
        <button
          class="c12"
          type="button"
        >
          <div
            class="c10"
          >
            <span
              class="c13"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c14"
      >
        <div
          class="c15 c16 c17 c18"
          spacing="2"
        />
      </div>
    </div>
  </div>
  <div
    class="c19"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;

exports[`FolderCard renders and matches the snapshot 1`] = `
.c7 {
  position: relative;
  overflow: hidden;
  max-width: 100%;
}

.c8 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c17 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c11 {
  font-weight: 500;
  color: #32324d;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c0 {
  position: relative;
}

.c5 {
  background: #eaf5ff;
  color: #66b7f1;
  padding-top: 8px;
  padding-right: 12px;
  padding-bottom: 8px;
  padding-left: 12px;
  border-radius: 4px;
}

.c10 {
  max-width: 100%;
}

.c2 {
  background: #ffffff;
  padding-top: 12px;
  padding-right: 16px;
  padding-bottom: 12px;
  padding-left: 16px;
  border-radius: 4px;
  box-shadow: 0px 1px 4px rgba(33,33,52,0.1);
  cursor: pointer;
  cursor: pointer;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c4 > * {
  margin-left: 0;
  margin-right: 0;
}

.c4 > * + * {
  margin-left: 12px;
}

.c13 {
  right: 16px;
}

.c14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c15 > * {
  margin-left: 0;
  margin-right: 0;
}

.c15 > * + * {
  margin-left: 8px;
}

.c16 {
  position: absolute;
  top: 12px;
}

.c1 {
  height: 100%;
  left: 0;
  position: absolute;
  opacity: 0;
  top: 0;
  width: 100%;
}

.c1:hover,
.c1:focus {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 path {
  fill: currentColor;
}

.c12 {
  display: none;
}

.c9 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
  <div
    class="c0 "
    tabindex="0"
  >
    <button
      aria-hidden="true"
      aria-label="Folder 1"
      class="c1"
      tabindex="-1"
      type="button"
    />
    <div
      class="c2 c3 c4"
      cursor="pointer"
      spacing="3"
    >
      <div
        class="c5"
      >
        <svg
          class="c6"
          fill="none"
          height="1.125rem"
          viewBox="0 0 24 24"
          width="1.25rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.414 5H21a1 1 0 011 1v14a1 1 0 01-1 1H3a1 1 0 01-1-1V4a1 1 0 011-1h7.414l2 2z"
            fill="#212134"
          />
        </svg>
      </div>
      <h2
        class="c7 c8 c9"
        id="folder-1-title"
        overflow="hidden"
      >
        <button
          class="c10"
          type="button"
        >
          <div
            class="c8"
          >
            <span
              class="c11"
            >
              Pictures
            </span>
          </div>
        </button>
      </h2>
      <div
        class="c12"
      >
        <div
          class="c13 c14 c15 c16"
          spacing="2"
        />
      </div>
    </div>
  </div>
  <div
    class="c17"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
